plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeSplitManager.java

```diff
@@ -291,7 +291,12 @@ private List<DeltaLakeSplit> splitsForFile(
     private static String buildSplitPath(String tableLocation, AddFileEntry addAction)
     {
         // paths are relative to the table location and URL encoded
-        return tableLocation + '/' + URLDecoder.decode(addAction.getPath(), UTF_8);
+        StringBuilder path = new StringBuilder(tableLocation);
+        if (!tableLocation.endsWith("/")) {
+            path.append('/');
+        }
+        path.append(URLDecoder.decode(addAction.getPath(), UTF_8));
+        return path.toString();
```

Comment:
I think this is a little nicer without the `StringBuilder` (but I don't particularly mind it as-is).

```java
String path = URLDecoder.decode(addAction.getPath(), UTF_8);
if (tableLocation.endsWith("/")) {
    return tableLocation + path;
}
return tableLocation + "/" + path;
```